File: week-3/requestment-2/go/main.go
package main

import (
	. "Project/functions"
	"fmt"
	"os"
)

func main() {
	numPages := 10
	url := "https://www.ptt.cc/bbs/movie/index.html"

	categories := [][]string{{}, {}, {}}

	for i := 0; i < numPages; i++ {
		page := GetPage(url)
		url = GetPreviousPageUrl(page)
		titles := GetTitles(page)
		for _, title := range titles {
			index := ValidatePrefix(title)
			if index >= 0 {
				categories[index] = append(categories[index], title)
			}
		}
	}

	content := ""
	for _, category := range categories {
		for _, title := range category {
			content += title + "\n"
		}
	}

	fout, err := os.Create("movie.txt")
	if err != nil {
		fmt.Println("movie.txt", err)
		return
	}
	defer fout.Close()
	fout.WriteString(content)
}

File: week-3/requestment-2/py/main.py
from functions import *

numPages = 10
url = 'https://www.ptt.cc/bbs/movie/index.html'

categories = [[], [], []]
for i in range(numPages):
    page = getPage(url)
    url = getPreviousPageUrl(page)
    titles = getTitles(page)
    for title in titles:
        index = validatePrefix(title)
        if index >= 0:
            categories[index].append(title)

with open('movie.txt', 'w', encoding='utf-8') as f:
    for category in categories:
        for title in category:
            f.write(f'{title}\n')
